File: Java/Commandes/src/Produit.java
public class Produit {
    private int id;
    private String designation;
    private double prixUnit;
    private int qteStock;

    private static int staticID = 0;

    Produit(String designation, double prixUnit, int qteStock) {
        this.id = staticID + 1;
        this.designation = designation;
        this.prixUnit = prixUnit;
        this.qteStock = qteStock;
        staticID++;
    }

    int GetID() {
        return id;
    }

    String GetDesignation() {
        return designation;
    }

    void SetDesignation(String designation) {
        this.designation = designation;
    }

    double GetPrixUnit() {
        return prixUnit;
    }

    void SetPrixUnit(double prixUnit) {
        this.prixUnit = prixUnit;
    }

    int GetQteStok() {
        return qteStock;
    }

    void SetQteStock(int qteStock) {
        this.qteStock = qteStock;
    }

}

File: php/Sandbox/Gestion_02/classes/produit.php
<?php

class Produit
{
    private $id;
    private $designation;
    private $qteStock;
    private $prixUnit;

    private static $produitNumber = 0;

    function __construct($designation, $qteStock, $prixUnit)
    {
        $id = self::$produitNumber;
        $this->designation = $designation;
        $this->qteStock = $qteStock;
        $this->prixUnit = $prixUnit;
        self::$produitNumber++;
    }

    function __destruct()
    {
        self::$produitNumber--;
    }

    public function getID()
    {
        return $this->id;
    }

    public function getDesignation()
    {
        return $this->designation;
    }
    public function setDesignation($designation)
    {
        $this->designation = $designation;
    }

    public function getQteStock()
    {
        return $this->qteStock;
    }
    public function setQteStock($qteStock)
    {
        $this->qteStock = $qteStock;
    }

    public function getPtixUnit()
    {
        return $this->prixUnit;
    }
    public function setPrexUnit($prixUnit)
    {
        $this->prixUnit = $prixUnit;
    }
}
